sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java

```diff
@@ -113,12 +134,163 @@ void setUp() throws CertificateException, IOException {
+            final CommunicationContext headerContext = new CommunicationContext(
+                    new HttpApplicationInfo(multimap, transactionId, requestUri),
+                    new TransportInfo(
+                            Constants.HTTPS_SCHEME, null, null, null, null, Collections.singletonList(certificate)));
+
+            try (final Message message = new Message(
+                    CommunicationLog.Direction.INBOUND,
+                    CommunicationLog.MessageType.REQUEST,
+                    headerContext,
+                    messageStorage)) {
+                message.write(String.format(
+                                BASE_MESSAGE_STRING,
+                                "action",
+                                String.format(SEQUENCE_ID_METRIC_BODY_STRING, "9223372036854775808", "1"))
+                        .getBytes(StandardCharsets.UTF_8));
+            }
+            messageStorage.flush();
+
+            verify(this.testRunObserver, atLeastOnce()).invalidateTestRun(any(NumberFormatException.class));
+        }
+    }
+
+    /**
+     * Tests whether an MdibVersion close to causing an overflow of the long field storing MdibVersion
+     * is saved as a positive value.
+     *
+     * @param dir message storage directory
+     * @throws IOException          on io exceptions
+     * @throws CertificateException on certificate exceptions
+     */
+    @Test
+    public void testMdibVersionCloseToOverflow(@TempDir final File dir) throws IOException, CertificateException {
+        try (final MessageStorage messageStorage =
+                new MessageStorage(1, mock(MessageFactory.class), new HibernateConfigImpl(dir), this.testRunObserver)) {
+            final ListMultimap<String, String> multimap = ArrayListMultimap.create();
+
+            final String transactionId = "transactionId";
+            final String requestUri = "requestUri";
+
+            final X509Certificate certificate = CertificateUtil.getDummyCert();
+            final CommunicationContext headerContext = new CommunicationContext(
+                    new HttpApplicationInfo(multimap, transactionId, requestUri),
+                    new TransportInfo(
+                            Constants.HTTPS_SCHEME, null, null, null, null, Collections.singletonList(certificate)));
+
+            try (final Message message = new Message(
+                    CommunicationLog.Direction.INBOUND,
+                    CommunicationLog.MessageType.REQUEST,
+                    headerContext,
+                    messageStorage)) {
+                message.write(String.format(
+                                BASE_MESSAGE_STRING,
+                                "action",
+                                String.format(SEQUENCE_ID_METRIC_BODY_STRING, "9223372036854775807", "1"))
+                        .getBytes(StandardCharsets.UTF_8));
+            }
+            messageStorage.flush();
+
+            final MessageContent messageContent =
```

Comment:
This throws an IOException for me. If I use try-with-resources, the exception no longer occurs.

Reply:
It is true that there was a connection leak and I fixed it, but I didn't get an IOException here.